src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -2610,7 +2610,7 @@ finalize_deployment_transaction_execute (RpmostreedTransaction *transaction,
   (void) rpmostree_syscore_bump_mtime (sysroot, NULL);
 
   sd_journal_print (LOG_INFO, "Finalized deployment; rebooting into %s", checksum);
-  rpmostreed_reboot (cancellable, error);
+  rpmostreed_daemon_reboot (rpmostreed_daemon_get ());
```

Comment:
What happens if there is a systemd inhibitor lock that is blocking shutdown? Would `systemctl reboot` called in `idle_initiate_reboot()` just fail? If that's the case, then would we have an unlocked staged deployment  since we'd `unlink()` `_OSTREE_SYSROOT_RUNSTATE_STAGED_LOCKED` but not actually reboot? Will we return a success message to the client even if we fail to reboot?